leshan-server-core/src/main/java/org/eclipse/leshan/server/queue/impl/RequestSendingTask.java

```diff
@@ -0,0 +1,137 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Request sending task is a Runnable, which is responsible for the actual sending of a queue request. The queue request
+ * is sent asynchronously; upon receiving a response or an error the message is removed from queue and a new task is
+ * scheduled to process the next message from queue.
+ */
+class RequestSendingTask implements Runnable {
+    private static final Logger LOG = LoggerFactory.getLogger(RequestSendingTask.class);
+    private final ClientRegistry clientRegistry;
+    private final LwM2mRequestSender requestSender;
+    private final Executor processingExecutor;
+    private final String endpoint;
+    private final MessageStore messageStore;
+    private final ClientStatusTracker clientStatusTracker;
+    private Map<Long, ResponseContext> responseContextHolder;
+
+    /**
+     * Creates a new task which is responsible for sending a queue request.
+     *
+     * @param clientRegistry client registry
+     * @param requestSender sender to perform send on (delegate)
+     * @param queueProcessingExecutor executor to schedule subsequent task on
+     * @param clientStatusTracker tracks the status of the client
+     * @param messageStore holds queued messages for the client
+     * @param endpoint clients endpoint identifier
+     * @param responseContextHolder holds the callback instances of the users for processing any response or error
+     *                              from client.
+     */
+    RequestSendingTask(ClientRegistry clientRegistry, LwM2mRequestSender requestSender,
+            Executor queueProcessingExecutor,
+            ClientStatusTracker clientStatusTracker, MessageStore messageStore, final String endpoint,
+            Map<Long, ResponseContext> responseContextHolder) {
+        this.clientRegistry = clientRegistry;
+        this.requestSender = requestSender;
+        this.processingExecutor = queueProcessingExecutor;
+        this.clientStatusTracker = clientStatusTracker;
+        this.endpoint = endpoint;
+        this.messageStore = messageStore;
+        this.responseContextHolder = responseContextHolder;
+    }
+
+    @Override
+    public void run() {
+        final QueuedRequest firstRequest = messageStore.retrieveFirst(endpoint);
+
+        if (firstRequest != null) {
+            try {
+                // TODO send with COAP retries ok? See
+                // https://github.com/OpenMobileAlliance/OMA-LwM2M-Public-Review/issues/32
+                final DownlinkRequest<LwM2mResponse> downlinkRequest = firstRequest.getDownlinkRequest();
+                LOG.debug("Sending request: {}", downlinkRequest);
+                final Client client = clientRegistry.get(firstRequest.getEndpoint());
+                if (client == null) {
+                    // client not registered anymore -> ignore this request
+                    LOG.debug("Client {} not registered anymore: {}", firstRequest.getEndpoint(), downlinkRequest);
```

Comment:
Yes, here we have the unnecessary "firstRequest.getEndpoint()" call if debug is disabled.

I can change it to 
`LOG.debug("Client {} not registered anymore: {}", endpoint, downlinkRequest);`